cluster-autoscaler/core/scale_up.go

```diff
@@ -428,7 +428,7 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 
 		newNodes := bestOption.NodeCount
 
-		if context.MaxNodesTotal > 0 && len(nodes)+newNodes > context.MaxNodesTotal {
+		if context.MaxNodesTotal > 0 && len(nodes)+newNodes+len(upcomingNodes) > context.MaxNodesTotal {
```

Comment:
`len(upcomingNodes)` will eventually gets to 0 so this will slow down the scale up process a bit but will not stop.

Reply:
> `len(upcomingNodes)` will eventually gets to 0 so this will slow down the scale up process a bit but will not stop.

The next iteration (`--scan-interval`) will then resume scale-up. So, if your `--scan-interval` is long then, yes, this will take longer.  We currently default to 10s so it will be RSN.